DODownloadFactory.cs

```diff
@@ -64,5 +79,56 @@ private void SetCommonProperties(IDODownload downloadObj, DOFile file)
                 downloadObj.SetProperty(DODownloadProperty.TotalSizeBytes, file.SizeBytes);
             }
         }
+
+        private List<IDODownload> GetDownloads(DO_DOWNLOAD_ENUM_CATEGORY? enumCategory)
+        {
+            var downloads = new List<IDODownload>();
+            IDOManager manager = GetDOManager();
+            var category = IntPtr.Zero;
+            try
+            {
+                if (enumCategory.HasValue)
+                {
+                    category = Marshal.AllocCoTaskMem(Marshal.SizeOf<DO_DOWNLOAD_ENUM_CATEGORY>());
```

Comment:
>AllocCoTaskMem [](http://example.com/codeflow?start=39&length=14)

Is AllocCoTaskMem really required? It's an in-param with no transfer of buffer ownership. I would rather pass enumCategory.Value directly to manager.EnumDownloads if we can.

What are we using for interop for this parameter?

Reply:
Public docs says that declaring this param as a `ref <struct> <param-name>` and passing `ref <variable>` at the call site should work, but it didn't. DO client receives an all-zero struct in this case.